src/answers.sql

```diff
@@ -1,15 +1,125 @@
 -- Your answers here:
 -- 1
+SELECT
+    c.name,
+    count(s.name)
+FROM
+    countries c
+    RIGHT JOIN states s ON c.id = s.country_id
+GROUP BY
+    c.name;
 
 -- 2
+SELECT
+    count(*) AS employees_without_bosses
+FROM
+    employees
+WHERE
+    supervisor_id IS NULL;
 
 -- 3
+SELECT
+    c.name,
+    t.address,
+    t.count
+FROM
+    (
+        SELECT
+            o.country_id,
+            address,
+            count(o.id)
+        FROM
+            offices o
+            INNER JOIN employees e ON o.id = e.office_id
+        GROUP BY
+            o.id
+    ) t
+    INNER JOIN countries c ON t.country_id = c.id
+ORDER BY
+    t.count DESC
+LIMIT
+    5;
 
 -- 4
+SELECT
+    supervisor_id,
+    count(supervisor_id)
+FROM
+    employees
+GROUP BY
+    supervisor_id
+ORDER BY
+    count DESC
+LIMIT
+    3;
 
 -- 5
+SELECT
+    count(*) AS list_of_office
+FROM
+    states s
+    INNER JOIN offices o ON s.id = o.state_id
```

Comment:
What happens if we have 2 states _Colorado_ in different countries?

Reply:
It will don't work.